src/conduit-lwt-unix/conduit_lwt_unix_ssl.real.ml

```diff
@@ -52,22 +52,59 @@ module Client = struct
 
   let default_ctx = create_ctx ()
 
-  let connect ?(ctx = default_ctx) ?src ?hostname sa =
+  type verify = { hostname : bool; ip : bool }
+
+  let default_verify = { hostname = true; ip = false }
+
+  let verification { hostname; ip } = function
+    | None, _ when hostname -> invalid_arg "impossible to verify hostname"
+    | _, None when ip -> invalid_arg "impossible to verify ip"
```

Comment:
are exceptions the way to go? shouldn't this code be executed as soon as possible (before opening a socket etc.)?

Reply:
I'll try to push the check to be done as soon as possible.

Exceptions are used elsewhere so I thought it was reasonable to keep using them

Reply:
I've pushed this check before to the very beginning of connect, before any socket is created